NTupleMaker/interface/functionsCP.h

```diff
@@ -383,9 +399,24 @@ TLorentzVector ipVec_Lepton(const AC1B * analysisTree, int tauIndex, TString ch)
   TLorentzVector vec;
   vec.SetXYZT(0.,0.,0.,0.);
 
+  
 TVector3 vertex(analysisTree->primvertex_x,
 		    analysisTree->primvertex_y,
 		    analysisTree->primvertex_z);
+
+  /*
+  //Merijn: temporarily replace vertex with gen level info
+  TVector3 vertex;
+  for (unsigned int igen=0; igen<analysisTree->genparticles_count; ++igen) {
+    if (analysisTree->genparticles_pdgid[igen]==23||analysisTree->genparticles_pdgid[igen]==24||
+	analysisTree->genparticles_pdgid[igen]==25||analysisTree->genparticles_pdgid[igen]==35||analysisTree->genparticles_pdgid[igen]==36) {
+      vertex.SetX(analysisTree->genparticles_vx[igen]);
+      vertex.SetY(analysisTree->genparticles_vy[igen]);
+      vertex.SetZ(analysisTree->genparticles_vz[igen]);
+      break;
+    }
+  }
+  */
```

Comment:
I understand that switching to gen level quantities is useful for test purposes, but having entire sections of the code commented is quite messy.
If I may give a suggestion: it would be less chaotic to have different functions using either gen level or reco level quantities and then switching between the two functions with a boolean. Or having one function returning either gen or reco level quantities given a boolean. This would avoid having too many commented parts of the code, and leaving comments in the code just for explanation purposes.

As an example a function like the following could do:
TVector3 GetVertex(const AC1B* analysisTree, bool genlevel){
TVector3 vertex(0.,0.,0.);
if(genlevel){
     for(....){
         if(.....) vertex.SetXYZ(analysisTree->genparticles_vx[igen],....)
     }
}else vertex.SetXYZ(analysisTree->primvertex_x,...);
return vertex;
}

Reply:
Hi Andrea,
if you want to add a boolean to the function to specify if it should take gen or reco vertex (take reco per default then please for RECO func), this is fine with me. 
If you specify a default you also conveniently per default you don't need to change the call of the function elsewhere in the code..
This could work

Reply:
Ok, I will take care of it as a lower priority thing to do. At some point it will be good to clean up the code for maintenance, that might be a good moment to do this.

Reply:
Hi Andrea, this sounds perfect. I anticipate to use it in the near future; once done we can remove it, or put as commented code in some graveyard somewhere.. Thanks!